src/jni_core/src/hummock_iterator.rs

```diff
@@ -105,6 +105,7 @@ impl HummockJavaBindingIterator {
             state_store_metrics: Arc::new(global_hummock_state_store_metrics(
                 MetricLevel::Disabled,
             )),
+            use_new_object_prefix_strategy: false,
```

Comment:
Let's make `use_new_object_prefix_strategy` part of `ReadPlan` instead of hardcoding it to false.